src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,97 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] arrayList;
+    private int size;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
+    private T[] grow() {
+        return Arrays.copyOf(arrayList, arrayList.length + arrayList.length / 2);
+    }
+
+    private boolean isIndexValid(int index) {
+        if (index >= 0 && index <= size) {
+            return true;
+        }
+        throw new ArrayIndexOutOfBoundsException("Wrong index. The index must be between"
+                + " 0 and size: " + size);
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == arrayList.length - 1) {
+            arrayList = grow();
+        }
+        arrayList[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        isIndexValid(index);
+        if (size == arrayList.length) {
+            arrayList = grow();
+        }
+        System.arraycopy(arrayList, index, arrayList, index + 1, size - index);
+        arrayList[index] = value;
+        size++;
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isIndexValid(index + 1);
```

Comment:
let's make two checks: One for methods where index should be in range : index >= 0 && index < size(get, set, remove);
And the second one: index >= 0 && index <= size
 which you will use only in add method(le'ts not make a method for this one)